server/app/repository/VersionRepository.java

```diff
@@ -452,6 +455,20 @@ private void validateProgramQuestionState() {
     }
   }
 
+  /** Validate there are no duplicate questions. */
+  private void validateNoDuplicateQuestions(ImmutableList<Long> questionList) {
```

Comment:
**suggestion (blocking):** since you've got the question definitions available at the call site, this could be a slightly more robust validation by checking for duplicates based on the `name` of the question rather than ID.

Reply:
Sounds good, updated!